server/storage_handler.go

```diff
@@ -478,17 +478,21 @@ func (s *storageWriteServer) appendRows(req *storagepb.AppendRowsRequest, msgDes
 			break
 		}
 	} else {
-		s, exists := s.streamMap[streamName]
+		st, exists := s.streamMap[streamName]
 		if !exists {
+			s.mu.RUnlock()
 			return fmt.Errorf("failed to get stream from %s", streamName)
 		}
-		status = s
+		status = st
 	}
 	s.mu.RUnlock()
 	if status.finalized {
 		return fmt.Errorf("stream is already finalized")
 	}
-	offset := req.GetOffset().Value
+	offset := int64(0)
```

Comment:
I'm not sure if using 0 here is good practice, though. I only ran into this because the Java client will not send an offset to just write to the end of a stream.